src/ResourceManagement/MachineLearning/Microsoft.Azure.Management.MachineLearning/Management/WebService.cs

```diff
@@ -0,0 +1,273 @@
+
+namespace Microsoft.Azure.MachineLearning
+{
+
+    /// <summary>
+    /// Describes an AzureML Web Service.
+    /// </summary>
+    public class WebService
+    {
+        /// <summary>
+        /// Internal reference to a AzureMLWebServicesManagementClient.
+        /// </summary>
+        private AzureMLWebServicesManagementClient _client { get; set; }
+
+        /// <summary>
+        /// Internal autogenerated model that holds properties of a web service.
+        /// TODO: Call it webservicedefinition in the swagger
+        /// </summary>
+        internal Management.MachineLearning.WebServices.Models.WebService Definition { get; set; }
+
+        /// <summary>
+        /// The name of this web service.
+        /// </summary>
+        public string Title
+        {
+            get { return this.Definition.Properties.Title; }
+            set
+            {
+                this.Definition.Properties.Title = value;
+                this.Definition.Name = value;
+            }
+        }
+
+        /// <summary>
+        /// The description for this web service.
+        /// </summary>
+        public string Description
+        {
+            get { return this.Definition.Properties.Description; }
+            set { this.Definition.Properties.Description = value; }
+        }
+
+        /// <summary>
+        /// DateTime for when this web service was created. Is readonly.
+        /// </summary>
+        public DateTime? CreatedOn
+        {
+            get
+            {
+                try
+                {
+                    var val = this.Definition.Properties.CreatedOn.Value;
+
+                    return val;
+                }
+                catch (Exception)
+                {
+                    return null;
+                }
+            }
+        }
+
+        /// <summary>
+        /// DateTime for when this web service was last modified. Is readonly.
+        /// </summary>
+        public DateTime? ModifiedOn
+        {
+            get
+            {
+                try
+                {
+                    var val = this.Definition.Properties.ModifiedOn.Value;
+
+                    return val;
+                }
+                catch (Exception)
+                {
+                    return null;
+                }
+            }
+
+            private set { throw new InvalidOperationException("Tried to set WebService.ModifiedOn. Cannot set value."); }
+        }
+
+        /// <summary>
+        /// Provisioning state of this web service. One of 'Unknown', 'Provisioning', 'Succeeded', 'Failed', or 'Canceled'.
+        /// </summary>
+        public string ProvisioningState 
+        {
+            // TODO: CHECK:InvalidOperationException if this is unknown
+            get
+            {
+                if (this.Definition.Properties.ProvisioningState.Equals("Unknown"))
+                {
+                    throw new InvalidOperationException();
```

Comment:
Throw exception from getter is also uncommon. Since "unknown" is a valid state, I think you could just return it.